toolkit/tools/internal/file/file.go

```diff
@@ -245,27 +262,56 @@ func GetAbsPathWithBase(baseDirPath, inputPath string) string {
 	return filepath.Join(baseDirPath, inputPath)
 }
 
+func IsDirEmpty(path string) (bool, error) {
+	contents, err := os.ReadDir(path)
+	if err != nil {
+		return false, fmt.Errorf("failed to check if dir is empty (%s):\n%w", path, err)
+	}
+
+	empty := len(contents) == 0
+	return empty, nil
+}
+
 // copyWithPermissions copies a file from src to dst, creating directories with the requested mode for the destination if needed.
 // Depending on the changeMode parameter, it may also change the file mode.
-func copyWithPermissions(src, dst string, dirmode os.FileMode, changeMode bool, filemode os.FileMode) (err error) {
+func copyWithPermissions(src, dst string, dirmode os.FileMode, changeMode bool, filemode os.FileMode,
+	noDereference bool,
+) (err error) {
 	const squashErrors = false
 
 	logger.Log.Debugf("Copying (%s) -> (%s)", src, dst)
 
```

Comment:
nit: could probably make two variables here, `isFile` and `isSymlink`. If-else would read better imo

Reply:
Duplicate comment.